utilities/os_utils.py

```diff
@@ -240,3 +241,67 @@ def generate_instance_type_rhel_os_matrix(preferences: list[str]) -> list[dict[s
         instance_types.append({preference: preference_config})
 
     return instance_types
+
+
+def generate_instance_type_centos_os_matrix(preferences: list[str]) -> list[dict[str, dict[str, Any]]]:
```

Comment:
duplicate code, please re-use `generate_instance_type_rhel_os_matrix`; remove `rhel` from the name and add an arg with the os type